test/conformance/ingress/util.go

```diff
@@ -844,12 +845,16 @@ func CreateDialContext(t *testing.T, ing *v1alpha1.Ingress, clients *test.Client
 }
 
 type RequestOption func(*http.Request)
+type Expectations struct {
+	HTTPResponseStatuses sets.Int
+	AllowDialError       bool
+}
 
 func RuntimeRequest(t *testing.T, client *http.Client, url string, opts ...RequestOption) *types.RuntimeInfo {
-	return RuntimeRequestWithStatus(t, client, url, sets.NewInt(http.StatusOK), opts...)
+	return RuntimeRequestWithStatus(t, client, url, Expectations{HTTPResponseStatuses: sets.NewInt(http.StatusOK)}, opts...)
 }
 
-func RuntimeRequestWithStatus(t *testing.T, client *http.Client, url string, expectedStatus sets.Int, opts ...RequestOption) *types.RuntimeInfo {
+func RuntimeRequestWithStatus(t *testing.T, client *http.Client, url string, expectations Expectations, opts ...RequestOption) *types.RuntimeInfo {
```

Comment:
One way to allow multiple types of options is to use `...interface{}` and then partition the allowed types inside the call, e.g.
```
for _, uto := range opts {
  switch to := uto.(type) {
    case RequestOption:
      reqOpts = append(reqOpts, to)
    case Expectation:
      expOpts = append(expOpts, to)
  }
}
```

Reply:
We can do that as well. Usually the expectations are all or nothing kind — no reason to continue, once the first one failed, while opts usually apply all.